### PR TITLE
Add js for add and remove lists

### PR DESCRIPTION
**Pull request by Tadessejemal1**
Project requirement 
-Remove all hardcoded items from the tasks array.
-Create a new JavaScript file for the new functionality.
-Implement a function for adding a new task (add a new element to the array).
-Implement a function for deleting a task (remove an element from the array).
-Implement a function for editing task descriptions.
-Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1)..
-All changes to the To Do List should be saved in local storage.

### CODE REVIEW

**Comment by Tadessejemal1** [comment]
When We Add the todo-list
 
image


When I clicked the checkbox the checked one line over the text
 
image


When I clicked the deleted icon, the list data is removed. 

 
image


The added data are stored on the local storage.
 
image

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,151 @@
+// /* eslint-disable no-unused-expressions */
+// expect(someTrueValue).to.be.true;
+
```
Comment (cynthiainga) [inline src/index.js:3]: - [ ] Please make sure to delete the comments lines of code in your file to keep a clean codebase 🙃🙃

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,151 @@
...
+        value: index === EditTodoId ? todoValue : todo.value,
+      }));
+      EditTodoId = -1;
+    } else {
+      todos.push(
+        {
+          value: todoValue,
+          checked: false,
+          color: 'orange',
+        },
```
Comment (cynthiainga) [inline src/index.js:91]: - [ ] All the tasks should have the property index set to the value of the new array length (i.e. if a 4th task is added to the list, the index of that task should equal 4) but for you, there is no property of index 🙃 🙃, please kindly fix that.

This is the look of your data in the local storage:
Screen Shot 2022-08-12 at 13 59 31

- [ ] You suppose to have a property of index and not of color 😁 😁

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,151 @@
...
+//     todosListEl.forEach(i => {
+//         if(i.classList.contains('check')){
+//             deleteTodo(i)
+//         }
+//     });
+//     let count = 0;
+//     const data = todos.from(localData).filter(i => !i.completed);
+//     data.map(i => i.index = count += 1);
+//     localStorage.setItem('todos', JSON.stringify(data));
+// }
```
Comment (cynthiainga) [inline src/index.js:149]: - [ ] Please make sure to delete the comments lines of code in your file to keep a clean codebase 🙃🙃

File: src/template.html [hunk 4]
```diff
@@ -6,29 +6,38 @@
...
         </div>
+      </header>
+      <div id="todos-list">
+        <!-- <div class="todo" id="0">
+          <i class="bi bi-circle"></i>
+          <i class="bi bi-check-circle-fill"></i>
+          <p class="">Go get milk.</p>
+          <i class="bi bi-pencil-square"></i>
+          <i class="bi bi-trash"></i>
+        </div> -->
```
Comment (cynthiainga) [inline src/template.html:35]: - [ ] Same remarque about the comments line of code, please make sure to delete the comments lines of code in your file to keep a clean codebase 🙃🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Tadessejemal1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- [x] No linters ✔️ 
- [x] Used the correct Github Flow 👍 
- [x] Descriptive pull request 👍 ✔️ 

## Required Changes ♻️

- [x] Please kindly check the comments under the review 👇 👇


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tadessejemal1** [comment]
Ias the previous reviewers mentioned I can fix the problems by adding an id for each value. this is a screenshot.
image

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @Tadessejemal1               :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for making all the changes requested by a previous reviewer good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

- [ ] [OPTIONAL] Your project was failing to start from my end and was throwing the error ```Missing script: "start"```. On further inspection, I realized that you do not have the "start" script to enable someone to run your project from their local machines. To fix this, you simply need to add ```    "start": "webpack serve --open", ``` after "dev" and it will work perfectly.

**This is the error it throws without it:**

image

**This is the correct scripts:**

```
"scripts": {
    "build": "webpack",
    "start": "webpack serve --open",
    "dev": "webpack serve"
  },
```

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**